### ISSUE TITLE
Build error for some components after update 

### ISSUE THREAD

**Issue by silveltman** [CloudCannon/bookshop#97]
I just updated bookshop from 2.4 > 2.6 and run into this error for some of my components: `TypeError: Converting circular structure to JSON`

I ran into this before, when calling a component inside a forloop (to which I found a workaround: https://github.com/CloudCannon/bookshop/issues/79). For this one however, I can't seem to recognise a pattern, therefore a clear cause. It seems to me quiet random. 

For instance, a simple component like this doesn't work

_map.eleventy.liquid_
```
<div class="map">
  <iframe src="{{ url }}"></iframe>
</div>
```

_page.liquid_
` {% bookshop "map" url: "test" %}
`
while other, more complex, components do still work. 

Site (in cloudcannon): https://quiet-beech.cloudvent.net/
In the following files I commented out the components that broke:  `routes.liquid`, `aanbod.liquid`, `carousel.eleventy.liquid`

**Comment by silveltman** [comment]
just updated, but still seeing the same issues 😬

**Comment by silveltman** [comment]
@bglw Can this issue be opened again please?

**Comment by bglw** [comment]
Apologies, certainly! I'm working on a Bookshop 3.0 release soon, so I'll look at including a fix there.

**Comment by bglw** [comment]
👋 Hi @silveltman 

Firstly, Bookshop 3.0 is out — see the [Migration Guide](https://github.com/CloudCannon/bookshop/blob/main/guides/migration.adoc) and [Release Notes](https://github.com/CloudCannon/bookshop/releases/tag/v3.0.0) 🙂

I had a look at those components, and it looks like the main issue is around the templating in your layout transforming the data before passing it to the Bookshop component.

I have added some discussion around this to the new documentation here: [Passing Data to Bookshop Components](https://github.com/CloudCannon/bookshop/blob/main/guides/eleventy.adoc#passing-data-to-bookshop-components). Essentially, part of this error is that any reassignments in your layout can't be known once the component is in the visual editor, so it doesn't have the data to re-render correctly.

Bookshop 3 now lets you access site data and collections directly from Eleventy components in the live editor, so moving more of that logic into your components will improve things. 

Additionally, there was a bug here causing the `TypeError: Converting circular structure to JSON` crash in your Eleventy build — this has been fixed in 3.0 😄 

I'll leave this issue open to cover any other issues you face migrating to 3.0
